zap.go

```diff
@@ -24,16 +31,30 @@ import (
 //   1. A time package format string (e.g. time.RFC3339).
 //   2. A boolean stating whether to use UTC time zone or local.
 func Ginzap(logger *zap.Logger, timeFormat string, utc bool) gin.HandlerFunc {
+	return GinzapWithConfig(logger, &Config{TimeFormat: timeFormat, UTC: utc})
+}
+
+// GinzapWithConfig returns a gin.HandlerFunc using configs
+func GinzapWithConfig(logger *zap.Logger, conf *Config) gin.HandlerFunc {
+	skipPaths := make(map[string]bool, len(conf.SkipPaths))
+	for _, path := range conf.SkipPaths {
+		skipPaths[path] = true
+	}
+
 	return func(c *gin.Context) {
 		start := time.Now()
 		// some evil middlewares modify this values
 		path := c.Request.URL.Path
 		query := c.Request.URL.RawQuery
 		c.Next()
 
+		if skipPaths[path] {
```

Comment:
can't return if match the skip paths.

See: https://github.com/gin-contrib/logger/blob/55c0f3c3242cfb91c355b151caaf8092528ff03b/logger.go#L42-L102

Reply:
Sorry, I can't understand why we can't return.

I think gin-contrib/logger support `SkipPathRegexp` so we can't return.
But this changes doesn't support regexp so we can return.
Does this change require regexp feature? (I think it is not difficult :) )

Reply:
@ota42y could you mirror the implementation in the main Gin logger here?

https://github.com/gin-gonic/gin/blob/34ce2104cad324f444943c528746bf6d23643cd3/logger.go#L243

This function should be pretty similar to that when it comes to the skip behavior.

Reply:
Fixed 🙋